atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/CassandraAtlasDbFactory.java

```diff
@@ -58,14 +64,34 @@ public KeyValueService createRawKeyValueService(
     @VisibleForTesting
     static CassandraKeyValueServiceConfig preprocessKvsConfig(
             KeyValueServiceConfig config,
+            Supplier<Optional<KeyValueServiceRuntimeConfig>> runtimeConfig,
             Optional<String> namespace) {
         Preconditions.checkArgument(config instanceof CassandraKeyValueServiceConfig,
-                "CassandraAtlasDbFactory expects a configuration of type"
-                        + " CassandraKeyValueServiceConfig, found %s", config.getClass());
+                "CassandraAtlasDbFactory expects an instance of type %s, found %s",
+                CassandraKeyValueServiceConfig.class, config.getClass());
         CassandraKeyValueServiceConfig cassandraConfig = (CassandraKeyValueServiceConfig) config;
 
         String desiredKeyspace = OptionalResolver.resolve(namespace, cassandraConfig.keyspace());
-        return CassandraKeyValueServiceConfigs.copyWithKeyspace(cassandraConfig, desiredKeyspace);
+        CassandraKeyValueServiceConfig configWithNamespace = CassandraKeyValueServiceConfigs
+                .copyWithKeyspace(cassandraConfig, desiredKeyspace);
+
+        return new CassandraReloadableKvsConfig(configWithNamespace, runtimeConfig);
+    }
+
+    @VisibleForTesting
+    Supplier<CassandraKeyValueServiceRuntimeConfig> preprocessKvsRuntimeConfig(
+            Supplier<Optional<KeyValueServiceRuntimeConfig>> runtimeConfig) {
+        return () -> {
+            Optional<KeyValueServiceRuntimeConfig> configOptional = runtimeConfig.get();
+
+            return configOptional.map(config -> {
```

Comment:
Does this mean that if I change the runtime config to something invalid (or invalid but of a different type), that the system will fall back to the default config?  Can we log a very clear message and keep the previous version instead?  Seems safer than potentially changing everything.